src/stream/src/common/table/state_table.rs

```diff
@@ -118,6 +176,12 @@ pub struct StateTableInner<
     watermark_buffer_strategy: W,
     /// State cleaning watermark. Old states will be cleaned under this watermark when committing.
     state_clean_watermark: Option<ScalarImpl>,
+
+    /// Watermark col idx
+    watermark_col_idx: Option<usize>,
```

Comment:
Wonder why we don't store this statically? 
Currently it seems like we store it in the watermark instead. Which seems strange to me.

```

#[derive(Debug, PartialEq, Eq, Clone)]
pub struct Watermark {
    pub col_idx: usize, // <---- why store here?
    pub data_type: DataType,
    pub val: ScalarImpl,
}
```